### PR TITLE
add support for hex and grid layout for DeblendedGalaxy

### PR DESCRIPTION
**Pull request by mr-superonion**
see issue https://github.com/LSSTDESC/descwl-shear-sims/issues/152

The galaxy number density is determined by the layout for these two cases.

### CODE REVIEW

**Comment by mr-superonion** [comment]
```
/usr/share/miniconda/envs/test/lib/python3.10/site-packages/hexalattice/hexalattice.py:17: in <module>
    matplotlib.use('Qt5Agg')
```
The test does not pass. But this is the problem of hexalattice. Do we want to fork that repo?

**Comment by mr-superonion** [comment]
It is this commit:
https://github.com/alexkaz2/hexalattice/commit/14bb67cf939c29b9d57e6b11815beea3a46e0218

**Comment by esheldon** [comment]
Yeah, that's a bug in hexalattice, it shouldn't require that

**Comment by mr-superonion** [comment]
@esheldon , I think the bug is fixed after the issue in hexalattice is fixed! Could you please review the PR?

File: descwl_shear_sims/galaxies.py [hunk 1]
```diff
@@ -599,14 +599,19 @@ def __init__(self, *, rng, coadd_dim, buff=0, layout='random'):
...
         if layout == 'random':
+            # need to calculate number of objects first
             # this layout is random in a square
             if (coadd_dim - 2*buff) < 2:
                 warnings.warn("dim - 2*buff <= 2, force it to 2.")
                 area = (2**SCALE/60)**2.
             else:
                 area = ((coadd_dim - 2*buff)*SCALE/60)**2
-
+            print("The area is of the simulation is %.2f arcmin2" % area)
```
Comment (esheldon) [inline descwl_shear_sims/galaxies.py:609]: remove print

File: descwl_shear_sims/galaxies.py [hunk 2]
```diff
@@ -616,14 +621,17 @@ def __init__(self, *, rng, coadd_dim, buff=0, layout='random'):
                 radius = (coadd_dim/2. - buff)*SCALE/60
                 area = np.pi*radius**2
             del radius
+            print("The area is of the simulation is %.2f arcmin2" % area)
```
Comment (esheldon) [inline descwl_shear_sims/galaxies.py:624]: remove print

**Review by esheldon — CHANGES_REQUESTED** [review]

**Comment by mr-superonion** [comment]
@esheldon , similar problem in stack, I think the problem comes with  `0.2023.29`. For ` 0.2023.28`, I did not see the error in the pytests.

**Review by esheldon — CHANGES_REQUESTED** [review]
these changes to the defaults are going to break existing configurations, making the code not backward compatible

**Comment by mr-superonion** [comment]
shall we merge?

**Pull request merged**